test/extensions/filters/http/transformation/inja_transformer_replace_test.cc

```diff
@@ -0,0 +1,491 @@
+  std::string body("not json body");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("not BAZ body", res);
+}
+
+TEST(Extraction, ReplaceWithSubgroupUnset) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  extractor.set_regex(".*(not (json) body)");
+  // subgroup is unset
+  auto replacement_text = "BAZ";
+	extractor.mutable_replacement_text()->set_value(replacement_text);
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::SINGLE_REPLACE);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("not json body");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("BAZ", res);
+}
+
+TEST(Extraction, ReplaceNoMatch) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  extractor.set_regex("this will not match the input string");
+  extractor.set_subgroup(0);
+  auto replacement_text = "BAZ";
+	extractor.mutable_replacement_text()->set_value(replacement_text);
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::SINGLE_REPLACE);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("not json body");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("", res);
+}
+
+TEST(Extraction, ReplacementTextLongerThanOriginalString) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  extractor.set_regex(".*(body)");
+  extractor.set_subgroup(1);
+  auto replacement_text = "this is a longer string than the original";
+  extractor.mutable_replacement_text()->set_value(replacement_text);
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::SINGLE_REPLACE);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("not json body");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("not json this is a longer string than the original", res);
+}
+
+TEST(Extraction, NilReplace) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  extractor.set_regex(".*(body)");
+  extractor.set_subgroup(1);
+  auto replacement_text = "";
+	extractor.mutable_replacement_text()->set_value(replacement_text);
```

Comment:
nit spacing?

Reply:
multiple places in this file

Reply:
ugh this is a tabs vs. spaces thing, thanks for pointing this out

Reply:
Resolved here: e8f9e74fbd152723852eba90849fa6e58bed0daf